app/models/course.rb

```diff
@@ -3,32 +3,26 @@ class Course < ActiveRecord::Base
   has_many   :sections, dependent: :destroy
   validates  :number, presence: true, uniqueness: { scope: :department_id }
   default_scope { order(number: :asc) }
+  
+  searchable do #searchable block required by sunspot (specifies which fields to index and search on)
+    text :number, :name, :description
+  end
 
   def self.get code, number
     joins(:department).where("departments.code = ? AND number = ?", code, number).first
   end
 
+  #Returns the result of a string search using sunspot_solr (formerly using SQL Query)
+  #params: a list of string search terms (split by whitespace from user search text)
   def self.search params
-    search_params = params.join(' & ')
-    query = <<-SQL
-      SELECT * FROM (
-        SELECT DISTINCT
-          courses.*,
-          to_tsvector(departments.name) ||
-          to_tsvector(departments.code) ||
-          to_tsvector(to_char(courses.number, '9999')) ||
-          to_tsvector(courses.name) ||
-          to_tsvector(coalesce((string_agg(array_to_string(sections.instructors, ' '), ' ')), ''))
-        AS document FROM courses
-        JOIN sections on sections.course_id = courses.id
-        JOIN departments on courses.department_id = departments.id
-        GROUP BY courses.id, sections.id, departments.id
-      ) c_search
-      WHERE c_search.document @@ to_tsquery('#{search_params}')
-      ORDER BY ts_rank(c_search.document, to_tsquery('#{search_params}')) DESC
-      LIMIT 25;
-    SQL
-    courses = find_by_sql(query).uniq
+    courses = Sunspot.search(Course) do
+      fulltext params do
+	    fields(:description, :number => 10, :name => 5.0) #weight results most strongly by number, then name, and finally description
+	    phrase_fields :name => 3.0 #weight titles that contain a phrase of search terms more strongly
+	  end
+	  paginate :page => 1, :per_page => 25 #place a maximum of 25 results on each page (for now we just display page 1)
+    end.results.uniq #store unique search results in courses in one step
```

Comment:
I noticed that we were calling uniq after the original SQL query, so I included it here as well, but I'm not entirely sure what constitutes unique results. Does our database have some identical duplicate class data in it (same exact name, number of credits, description, etc..)? The closest thing Sunspot has may be grouping/field collapsing http://wiki.apache.org/solr/FieldCollapsing